sw/device/tests/sim_dv/inject_scramble_seed.c

```diff
@@ -23,29 +25,98 @@
 
 static dif_rstmgr_t rstmgr;
 static dif_otp_ctrl_t otp_ctrl;
+static dif_flash_ctrl_state_t flash_ctrl;
 
-enum { kFlashWordSize = FLASH_CTRL_PARAM_BYTES_PER_WORD };
+enum {
+  kFlashWordSize = FLASH_CTRL_PARAM_BYTES_PER_WORD,
+  kFlashPageSize = FLASH_CTRL_PARAM_BYTES_PER_PAGE,
+  kFlashStartAddr = TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR,
+  kFlashMpRegions = FLASH_CTRL_PARAM_NUM_REGIONS
+};
 
 OTTF_DEFINE_TEST_CONFIG();
 
+// Expected data for the scramble seed, directly backdoor loaded by the
+// test environment.  The data has to be backdoor loaded because the
+// life cycle state used in this test does not permit the isolated
+// partition to be written.
+static volatile const uint32_t kIsoPartExpData[16];
+
+static bool check_iso_data(dif_flash_ctrl_state_t *flash_ctrl) {
```

Comment:
nit: This could also be defined as a `void` function if all the checks generate assertions.

Reply:
updated